src/Adapter/Carrier/Validate/CarrierValidator.php

```diff
@@ -102,4 +98,43 @@ public function validateGroupsExist(array $groupIds): void
             $this->groupRepository->assertGroupExists(new GroupId((int) $groupId));
         }
     }
+
+    /**
+     * @throws CoreException
+     */
+    private function validateGeneral(Carrier $carrier): void
+    {
+        $this->validateObjectModelProperty($carrier, 'name', CarrierConstraintException::class, CarrierConstraintException::INVALID_NAME);
+        $this->validateObjectModelProperty($carrier, 'grade', CarrierConstraintException::class, CarrierConstraintException::INVALID_GRADE);
+        $this->validateObjectModelProperty($carrier, 'url', CarrierConstraintException::class, CarrierConstraintException::INVALID_TRACKING_URL);
+        $this->validateObjectModelProperty($carrier, 'position', CarrierConstraintException::class, CarrierConstraintException::INVALID_POSITION);
+        $this->validateObjectModelLocalizedProperty($carrier, 'delay', CarrierConstraintException::class, CarrierConstraintException::INVALID_DELAY);
+        $this->validateObjectModelProperty($carrier, 'max_width', CarrierConstraintException::class, CarrierConstraintException::INVALID_MAX_WIDTH);
+        $this->validateObjectModelProperty($carrier, 'max_height', CarrierConstraintException::class, CarrierConstraintException::INVALID_MAX_HEIGHT);
+        $this->validateObjectModelProperty($carrier, 'max_depth', CarrierConstraintException::class, CarrierConstraintException::INVALID_MAX_DEPTH);
+        $this->validateObjectModelProperty($carrier, 'max_weight', CarrierConstraintException::class, CarrierConstraintException::INVALID_MAX_WEIGHT);
+    }
+
+    /**
+     * @throws CoreException
+     */
+    private function validateShippingGeneral(Carrier $carrier): void
```

Comment:
```suggestion
    private function validateShipping(Carrier $carrier): void
```
Optional, can be done in another PR